src/containers/articles/NewArticle.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import ArticlesForm from '../../components/articles/ArticlesForm';
+import { createArticle, editArticle } from '../../actions/articleActions';
+import imageUploader from '../../utils/imageUploader';
+
+
+class CreateUpdateArticle extends Component {
+  state = {
+    title: '',
+    description: '',
+    body: '',
+    tagList: [],
+    image: '',
+  }
+
+  componentDidMount() {
+    const { create, articleData } = this.props;
+    if (!create) {
+      this.setState({
+        title: articleData.title,
+        description: articleData.description,
+        body: articleData.body,
+        tagList: articleData.tagList,
+        image: articleData.image,
+      });
+    }
+  }
+
+  handleChange = (event, element = 'title') => {
+    switch (element) {
+      case 'editor':
+        this.setState({
+          body: event.editor.getData(),
+        });
+        break;
+      case 'tags':
+        this.setState({
+          tagList: event,
+        });
+        break;
+      case 'image':
+        this.setState({
+          image: event,
+        });
+        break;
+
+      case 'description':
+        this.setState({
+          description: event.target.value,
+        });
+        break;
+      default:
+        this.setState({
+          title: event.target.value,
+        });
+        break;
+    }
+  }
+
+  handleSubmit = (event, create, slug = '') => {
+    event.preventDefault();
+    const { createArticle, editArticle, history } = this.props; //  eslint-disable-line
+    const {
+      title, description, body, tagList, image,
+    } = { ...this.state };
+    let data = {};
```

Comment:
This can be implemented better:

const data = image ? { title, description, body, tagList, image } : { title, description, body, tagList }